app/DynamicGraphs/WriteRunDot.hs

```diff
@@ -12,5 +12,8 @@ doDots cases = do
 createImage :: PrintDotRepr dg n => (FilePath, dg n) -> IO FilePath
 createImage (n, g) = createImageInDir (normalise "graphs/gen") n Png g
 
+-- (question)did not find fuction addExtension under Data.GraphViz
```

Comment:
`addExtension` is actually found under [Data.GraphViz.Commands](https://hackage.haskell.org/package/graphviz-2999.19.0.0/docs/Data-GraphViz-Commands.html#v:addExtension), but this is *re-exported* by `Data.GraphViz` so you only need to import the latter.